pkg/core/manager/workflow.go

```diff
@@ -101,9 +101,40 @@ func (mgr *workflowManager) UnassignFromCodeset(ctx context.Context, name, codes
 }
 
 func (mgr *workflowManager) ListAssignments(ctx context.Context, name *string) ([]*workflow.WorkflowAssignment, error) {
-	return nil, nil
+	assignments := []*workflow.WorkflowAssignment{}
+	for wf, acs := range mgr.workflowStore.GetAssignments(ctx, name) {
+
+		listener, err := mgr.workflowBackend.GetWorkflowListener(ctx, wf)
+		if err != nil {
+			return nil, err
+		}
+
+		assignment := newWorkflowAssignment(wf, acs, listener)
+		assignments = append(assignments, assignment)
+	}
+	return assignments, nil
 }
 
 func (mgr *workflowManager) ListRuns(ctx context.Context, filter *domain.WorkflowRunFilter) ([]*workflow.WorkflowRun, error) {
 	return nil, nil
 }
+
+func newWorkflowAssignment(workflowName string, codesets []*domain.AssignedCodeset, listener *domain.WorkflowListener) *workflow.WorkflowAssignment {
+	assignment := &workflow.WorkflowAssignment{
+		Workflow: &workflowName,
+		Status: &workflow.WorkflowAssignmentStatus{
+			Available: &listener.Available,
+			URL:       &listener.DashboardURL,
+		},
+	}
+	for _, c := range codesets {
+		assignment.Codesets = append(assignment.Codesets, &workflow.Codeset{
+			Name:        c.Codeset.Name,
+			Project:     c.Codeset.Project,
+			Description: c.Codeset.Description,
+			Labels:      c.Codeset.Labels,
+			URL:         &c.Codeset.URL,
+		})
+	}
+	return assignment
+}
```

Comment:
The reason you still need to do this conversion here is that the workflow manager and its sub-components still use the workflow data structures from outside the "domain", those generated by goa, while the codeset store has already been migrated to use "domain" structures and the service code is doing all the conversion.

I wanted to mention this later anyway, so here goes: the one thing that is missing to achieve full independence for the workflow manager is to do the same with workflow data structures: define "domain" data structures for workflows and assignments, and only use those in the workflow manager, store and back-end. Then update the service to make the conversion between goa generated data structures and "domain" ones when calling the manager interface.

But that is too much to ask from this PR, it's already doing an excellent job and it's already big enough. So I'll just say this: this is conversion code and belongs in the service rather than here.